src/Chapter/DataStructures.elm

```diff
@@ -4,10 +4,120 @@ data =
+
+```elm
+type Animal
+  = Rabbit String
+  | Dog String Int
+  | Cat String Int
+
+myBunny : Animal
+myBunny = Rabbit "white"
+
+cindysPuppy : Animal
+cindysPuppy = Dog "Cindy" 3
+
+johnsCat : Animal
+johnsCat = Cat "John" 2
+```
+<p class=note> You'll notice that instead of `number`, here we use `Int`.   `number` is actually a special type which can be either an `Int` (an integer value) or a `Float` (a decimal number). Elm normally decides for you which type `number` should be, but when defining union types, you need to specify which you'd like to use. In this case, we only want whole number ages, so we use `Int`.
+
+<p class=note>`Rabbit String`, `Dog String Int`, `Cat String Int` are all the different <dfn>type constructors</dfn> of Animal. They're called this because they are used to create or *construct* values of the type Animal.
+
+You can use union types in the way you would use any other type. You can make functions that operate on and with them (we'll get to that soon!), you can display them with Text.asText, you can even use them in other union types! Now that we know we can use union types in other union types, we can do some cool things. Let's try redefining Animal again to keep track of each Animal's age, if it's wild or domestic, and its owner if it is domestic. We'll do this using a second union type which we'll call Owner.
+
+```elm
+type Owner
+  = Domestic String
+  | Wild
+
+type Animal
+  = Rabbit Owner Int
+  | Dog Owner Int
+  | Cat Owner Int
+
+wildBunny : Animal
+wildBunny = Rabbit (Wild) 3
+
+myPuppy : Animal
+myPuppy = Dog (Domestic "Me") 8
+
+johnsCat : Animal
+johnsCat = Cat (Domestic "John") 7
+```
+
 # Pattern Matching with Case Expressions
 
+Now that we know what union types are and how to define them, let's look at creating some functions that make use of them. First, we'll define a function which checks if an Animal (a type we defined in the last section) is domestic or not.
+
+<p class="note"> In this example, we'll make use of a new type called `Bool` (short for Boolean). A Bool can be have a value of either `True` or `False` and as you might have guessed represent a statement being either true or false. In this case, the function will return `True` if the Animal is a cat, and `False` if the animal is a Dog or Rabbit.
+
+```elm
+import Text
+
+...
+
+isRabbit : Animal -> Bool
+isRabbit mysteryPet =
+  case mysteryPet of
+    Rabbit owner age -> True
+    Dog owner age -> False
+    Cat owner age -> False
+
+main = Text.asText (isRabbit johnsCat)
+
+```
+
+As you can see, the program outputs false. If you switch `johnsCat` with `wildBunny` in the top level declaration of main, you'll see the result changes to true, as expected.
```

Comment:
Perhaps we could do `animals = [johnsCat, myPuppy, wildBunny]` and then `map isRabbit animals]`?